x/auth/ante/sigverify.go

```diff
@@ -83,6 +83,15 @@ func (spkd SetPubKeyDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		if acc.GetPubKey() != nil {
 			continue
 		}
+
+		//Validate public key for bls12381 so that the public key only needs to be checked once
+		switch pubkey := pk.(type) {
+		case *bls12381.PubKey:
+			if !pubkey.Validate() {
```

Comment:
Just for my understanding does `.Valdiate()` memoize something, which is why calling it here speeds up the subsequent code?

Reply:
Ah, okay, then how come we only do this validation for the new public key type and not the other types of key?

Reply:
The other public keys don't care because their curves are very fast and the cost is very tiny so it is fine to do that repeatedly in the signature verification, also there is no quick interface for the validation.